### DISCUSSION TITLE
Add a file and immediately open as a buffer

### DISCUSSION THREAD

**Discussion by varunshoor** [nvim-neo-tree/neo-tree.nvim#718] [Q&A]
Hi,
I searched before posting but can't seem to find an answer.

I would like to open the newly added file immediately as a buffer so I can start writing my code, currently I have to open it from the tree after it gets created.

Is it possible with current options? I can't seem to find the relevant one in docs

Thanks!

**Comment by pysan3** [comment]
A solution that came up to my mind is to look at [this file](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/v2.x/lua/neo-tree/sources/filesystem/commands.lua) and combine few of them as a new command that suits your needs and map it to a new keybind.

Sorry, I haven't tested them but for example

```lua
command = {
  create_and_edit = function (state)
    local commands = require("neo-tree.sources.filesystem.commands")
    commands.add(state)
    commands.open(state)
  end,
}
```

**Reply by varunshoor** [reply]
Thanks for replying! The addition works but the file doesn't open. I think maybe `.open()` expects the file to be created and maybe there is a delay after creation which prevents the immediate opening?

**Comment by pysan3** [comment]
I thought we had to rely on neo-tree to handle the new file but it was way more simple.
@varunshoor 

```lua
command = {
  create_and_edit = function(state)
    local cc = require("neo-tree.sources.common.commands")
    local fs = require("neo-tree.sources.filesystem")
    cc.add(state, function(destination) -- callback to cc.add, destination is the new file/dir path
      fs.show_new_children(state, destination) -- lazily update neo-tree to show created item
      local file_info = vim.loop.fs_stat(destination)
      if file_info and file_info.type == "file" then -- ignoring when destination is dir
        vim.cmd.edit(destination)
      end
    end)
  end,
}
```
